source/change-streams.rst

```diff
@@ -282,28 +305,37 @@ Driver API
      */
     collation: Optional<Document>;
 
+    /**
+     * The change stream will only provides changes that occurred after the
+     * specified cluster time. Any command run against the server will return
+     * a $clusterTime field that can be used here. The default value is a
+     * cluster time obtained from the server before the change stream was created.
+     * @since 4.0
+     * @see https://docs.mongodb.com/manual/reference/method/db.runCommand/
+     */
+    startAtClusterTime: Optional<Document>;
```

Comment:
The server does not accept a `$clusterTime` here. The startAtClusterTime field must be used like this:
```python
>>> coll.aggregate([{'$changeStream': {'startAtClusterTime': {'ts': s.cluster_time['clusterTime']}}}])
<pymongo.command_cursor.CommandCursor object at 0x1035e1390>
```

Passing a clusterTime results in an error:
```python
>>> coll.aggregate([{'$changeStream': {'startAtClusterTime': s.cluster_time}}])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/shane/Library/Python/2.7/lib/python/site-packages/pymongo/collection.py", line 2181, in aggregate
    **kwargs)
  File "/Users/shane/Library/Python/2.7/lib/python/site-packages/pymongo/collection.py", line 2088, in _aggregate
    client=self.__database.client)
  File "/Users/shane/Library/Python/2.7/lib/python/site-packages/pymongo/pool.py", line 496, in command
    collation=collation)
  File "/Users/shane/Library/Python/2.7/lib/python/site-packages/pymongo/network.py", line 125, in command
    parse_write_concern_error=parse_write_concern_error)
  File "/Users/shane/Library/Python/2.7/lib/python/site-packages/pymongo/helpers.py", line 146, in _check_command_response
    raise OperationFailure(msg % errmsg, code, response)
pymongo.errors.OperationFailure: BSON field '$changeStream.startAtClusterTime.clusterTime' is an unknown field.
```

The server has stated elsewhere that do not want drivers peeking inside the contents of the `$clusterTime` document ([SERVER-31916](https://jira.mongodb.org/browse/SERVER-31916?focusedCommentId=1745844&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-1745844)). So I think what we should actually accept is an `operationTime`:
```python
>>> coll.aggregate([{'$changeStream': {'startAtClusterTime': {'ts': s.operation_time}}}])
<pymongo.command_cursor.CommandCursor object at 0x1035e1310>
```

I propose we rename this option (and hopefully the `$changeStream` field) to `startAtOperationTime` since it does not accept a cluster time.

Reply:
Nice catch, this is very confusing in the server design document because there are examples of providing just the timestamp _and_ the now required form `{ ts: Timestamp }`

Reply:
In this case, can we update the Server Design document?

Reply:
Yes, it should be updated. Please add a comment or suggestion to that document.